src/main/java/de/tum/in/www1/artemis/config/MetricsBean.java

```diff
@@ -450,7 +450,7 @@ public void updatePublicArtemisMetrics() {
 
         ensureCourseInformationIsSet(courses);
 
-        var activeCourses = courses.stream()
+        var activeCourses = courses.stream().filter(course -> !course.isTestCourse())
```

Comment:
Can you filter before to avoid the database calls to count the students for inactive course and for test courses? This should reduce the database load quite significantly?

Ideally we even filter this into the database query instead of using findAll

Reply:
You are right. I did not change that since the full courses list was used further down. It was not actually needed there, so the query has been optimised now to only fetch relevant courses.